src/SuperstateToken.sol

```diff
@@ -412,26 +316,74 @@ contract SuperstateToken is ISuperstateToken, ERC20Upgradeable, PausableUpgradea
      * @param src Source address from which tokens will be burned
      * @param amount Amount of tokens to burn
      */
-    function burn(address src, uint256 amount) external {
+    function adminBurn(address src, uint256 amount) external {
         _checkOwner();
         _requireNotAccountingPaused();
-        if (availableBalanceOf(src) < amount) revert InsufficientAvailableBalance();
 
         _burn(src, amount);
-        emit Burn(msg.sender, src, amount);
+        emit AdminBurn({burner: msg.sender, src: src, amount: amount});
+    }
+
+    /**
+     * @notice Burn tokens from the caller's address for offchain redemption
+     * @param amount Amount of tokens to burn
+     */
+    function offchainRedeem(uint256 amount) external {
+        _requireNotAccountingPaused();
+        if (!isAllowed(msg.sender)) revert InsufficientPermissions();
+
+        _burn(msg.sender, amount);
+        emit OffchainRedeem({burner: msg.sender, src: msg.sender, amount: amount});
     }
 
     /**
-     * @notice Burn tokens from the caller's address
+     * @notice Burn tokens from the caller's address to bridge to another chain
+     * @dev If destination address on chainId isn't on allowlist, or chainID isn't supported, tokens wind up in book entry
      * @param amount Amount of tokens to burn
+     * @param ethDestinationAddress ETH address to send to on another chain
+     * @param otherDestinationAddress Non-EVM addresses to send to on another chain
+     * @param chainId Numerical identifier of destination chain to send tokens to
      */
-    function burn(uint256 amount) external {
+    function bridge(uint256 amount, address ethDestinationAddress, string memory otherDestinationAddress, uint256 chainId) public {
         _requireNotAccountingPaused();
-        if (availableBalanceOf(msg.sender) < amount) revert InsufficientAvailableBalance();
-        if (!hasSufficientPermissions(msg.sender)) revert InsufficientPermissions();
+
+        if (!isAllowed(msg.sender)) { revert InsufficientPermissions(); }
+
+        if (amount == 0) {
+            revert ZeroSuperstateTokensOut();
+        }
+
+        if (ethDestinationAddress != address(0) && bytes(otherDestinationAddress).length != 0) {
```

Comment:
we should revert if both addresses are 0 and chainID is not zero right? or if either is nonzero and chain id is 0? maybe a comment here about that and in natspec about usage for book entry

Reply:
Probably worth doing the chain_id one too, will add